api/lanes.py

```diff
@@ -562,3 +590,59 @@ def apply_filters(self, qu, work_model=Work, *args, **kwargs):
         qu = qu.join(work_edition).filter(work_edition.series==self.series)
         qu = qu.order_by(work_edition.series_position, work_edition.title)
         return qu
+
+
+class ContributorLane(QueryGeneratedLane):
+    """A lane of Works written by a particular contributor"""
+
+    ROUTE = 'contributor'
+    MAX_CACHE_AGE = 48*60*60    # 48 hours
+
+    def __init__(self, _db, contributor_name, contributor_id=None):
+        if not contributor_name:
+            raise ValueError("ContributorLane can't be created without contributor")
+
+        self.contributor_name = contributor_name
+        self.contributor = None
+        if contributor_id:
+            self.contributor = get_one(_db, Contributor, id=contributor_id)
+            if (self.contributor_name!=self.contributor.display_name and
+                self.contributor_name!=self.contributor.name):
+                raise ValueError(
+                    "ContributorLane can't be created with inaccurate"
+                    " Contributor data."
+                )
+
+        full_name = display_name = "Books by %s" % self.contributor_name
```

Comment:
This string needs to be localized but that's a big job that can't go into this branch. I filed https://github.com/NYPL-Simplified/server_core/issues/358 to deal with it.